### PR TITLE
feat: Add proper tray icons for Windows dark mode

### PR DESCRIPTION
**Pull request by Phylu**
Implement a tray icon to be used on Windows when dark mode is switched on.

This implementation is windows only, as on Mac the icon color is automatically inverted and Linux ships with the icons used for a dark tray only.

Solves: https://community.getmailspring.com/t/dark-gray-icon-on-windows-dark-mode/7337

### CODE REVIEW

**Comment by 3v1n0** [comment]
FYI, in linux you can also get this info using the [color scheme portal setting](https://flatpak.github.io/xdg-desktop-portal/docs/doc-org.freedesktop.portal.Settings.html).

However, you also need to consider that under GNOME the preference won't change the shell unless is set to a value that is not exposed by such interface (to force light/dark).

**Comment by Phylu** [comment]
Yes, the problem is that a lot of Linux desktop environments have a dark tray even when a light theme is activated. The system color should be available similar to Windows using the electron module. But I think that any change here would make the icon less visible in most cases. Therefore, I did not touch this for now.

@3v1n0 However I am open for ideas if you have any.

**Comment by 3v1n0** [comment]
> Yes, the problem is that a lot of Linux desktop environments have a dark tray even when a light theme is activated. The system color should be available similar to Windows using the electron module. But I think that any change here would make the icon less visible in most cases. Therefore, I did not touch this for now.

True, sadly as said the system info is not provided to apps, even though it's available in gnome settings for example, but not something mailspring can check from a container.

In theory the way to do this properly would be to use symbolic icons, so that the DE can colorize them properly.

Now... the problem is that IIRC electron nowadays does not use `libappindicator` anymore so the icons will be pixbufs and these can't use this feature.

File: app/internal_packages/preferences/lib/tabs/preferences-appearance.tsx [hunk 1]
```diff
@@ -180,11 +178,12 @@ class TrayIconStylePicker extends React.Component<{ config: ConfigLike }> {
   };
 
   render() {
+    let systemTrayIconScore = new SystemTrayIconStore();
```
Comment (bengotow) [inline app/internal_packages/preferences/lib/tabs/preferences-appearance.tsx:181]: It feels a little unusual to instantiate a store in a render method, but as long as we don't call `activate` it looks like this doesn't have any impact and is a-ok. Let's ship it!  

(I think an alternative might be to make all of the "getPath" methods "static" methods of the class so you can call them like `SystemTrayIconStore.inboxFullUnreadIcon()`, but probably not worth changing here)

File: app/internal_packages/system-tray/lib/system-tray-icon-store.ts [hunk 2]
```diff
@@ -84,24 +67,62 @@ class SystemTrayIconStore {
...
   };
 
+  // This implementation is windows only.
+  // On Mac the icon color is automatically inverted
+  // Linux ships with the icons used for a dark tray only
+  _dark = () => {
+    if (nativeTheme.shouldUseDarkColors && process.platform === 'win32') {
+      return "-dark";
+    }
+    return "";
```
Comment (bengotow) [inline app/internal_packages/system-tray/lib/system-tray-icon-store.ts:77]: Ahh this is cool! It's kinda surprising that this isn't necessary on macOS too, but it looks like `isTemplateImg: true` makes the image auto-invert there.

**Review by bengotow — APPROVED** [review]

File: app/internal_packages/preferences/lib/tabs/preferences-appearance.tsx (hunk 1 above)
Comment (Phylu) [reply app/internal_packages/preferences/lib/tabs/preferences-appearance.tsx:181]: I am happy to refactor this in a future PR. I will see if I deem it useful and necessary.

**Pull request merged**